src/core/include/ngraph/variant.hpp

```diff
@@ -18,28 +18,77 @@ using ov::Node;
 using VariantTypeInfo = ov::DiscreteTypeInfo;
 
 using Variant = ov::RuntimeAttribute;
-template <typename T>
-using VariantImpl = ov::RuntimeAttributeImpl<T>;
 
-template <typename T>
-using VariantWrapper = ov::RuntimeAttributeWrapper<T>;
+template <typename VT>
+class OPENVINO_DEPRECATED("Please use ov::Any to store VT directly") VariantImpl : public Variant {
+public:
+    OPENVINO_RTTI(typeid(VT).name());
+    using value_type = VT;
+
+    VariantImpl() = default;
+
+    VariantImpl(const value_type& value) : m_value(value) {}
+
+    const value_type& get() const {
+        return m_value;
+    }
+    value_type& get() {
+        return m_value;
+    }
+    void set(const value_type& value) {
+        m_value = value;
+    }
+
+protected:
+    value_type m_value;
+};
+
+template <typename VT>
+class OPENVINO_DEPRECATED("Please use ov::Any to store VT directly") VariantWrapper {};
+
+OPENVINO_SUPPRESS_DEPRECATED_START
+extern template class OPENVINO_API VariantImpl<std::string>;
+extern template class OPENVINO_API VariantImpl<int64_t>;
+extern template class OPENVINO_API VariantImpl<bool>;
+
+template <>
+class OPENVINO_API VariantWrapper<std::string> : public VariantImpl<std::string> {
+public:
+    OPENVINO_RTTI("VariantWrapper<std::string>");
+    VariantWrapper(const value_type& value) : VariantImpl<value_type>(value) {}
+    std::string to_string() const override {
+        return m_value;
+    }
+};
+
+template <>
+class OPENVINO_API VariantWrapper<int64_t> : public VariantImpl<int64_t> {
```

Comment:
should not `OPENVINO_DEPRECATED` be used here as well explicitly?

Reply:
It inherits from deprecated class. Seems that it is enough.